# Python Questions

## Introduction to Python

### Question 1
Write a program to input marks of three tests of a student (all integers). Then calculate and print the average of all test marks.  

Sample Input 1 :  
3  
4  
6  

Sample Output 1 :  
4.333333333333333

In [1]:
a = int(input())
b = int(input())
c = int(input())
d = (a+b+c)/3
print(d)

4.333333333333333


### Question 2
You are given two integers: X and N. You have to calculate X raised to power N and print it.

**Input format:**
The first line of input contains an integer X (1 <= X <= 100)
The second line of input contains an integer N (1 <= N <= 10) 

**Constraints:**
Time Limit: 1 second

**Output format:**
The first and only line of output contains the result.  

Sample Input:  
10  
4  

Sample Output:  
10000

In [1]:
X = int(input())
N = int(input())

a = X**N

print(a)

10000


### Question 3
*Arithmetic Progression*

You are given first three entries of an arithmetic progression. You have to calculate the common difference and print it.

**Input format:**
The first line of input contains an integer a (1 <= a <= 100)  
The second line of input contains an integer b (1 <= b <= 100)   
The third line of input contains an integer c (1 <= c <= 100)  

**Constraints:**
Time Limit: 1 second

**Output format:**
The first and only line of output contains the result.  

Sample Input:  
1  
3  
5  
Sample Output:  
2

In [ ]:
a  = int(input())
b  = int(input())
c  = int(input())

d = c - b

print(d)

### Question 4
*Rectangular Area*

You are given a rectangle in a plane. The corner coordinates of this rectangle is provided to you. You have to print the amount of area of the plane covered by this rectangle.
The end coordinates are provided as four integral values: x1, y1, x2, y2. It is given that x1 < x2 and y1 < y2.

**Input format:**  
The first line of input contains an integer x1 (1 <= x1 <= 10)  
The second line of input contains an integer y1 (1 <= y1 <= 10)  
The third line of input contains an integer x2 (1 <= x2 <= 10)  
The fourth line of input contains an integer y2 (1 <= y2 <= 10)  

**Constraints:**
Time Limit: 1 second

**Output format:** 
The first and only line of output contains the result.  

Sample Input:  
1  
1  
3  
3  

Sample Output:  
4  

In [8]:
x1 = int(input())
x2 = int(input())
y1 = int(input())
y2 = int(input())

a = x1 - y1
b = x2 - y2
d = a * b

print(d)


4


## Conditionals and Loops

### Question 1

#### Check Number

Given an integer n, find if n is positive, negative or 0.
If n is positive, print "Positive"
If n is negative, print "Negative"
And if n is equal to 0, print "Zero".

Input Format :  
Integer n  
Output Format :  
"Positive" or "Negative" or "Zero" (without double quotes)

**Constraints :**
1 <= n <= 100  

Sample Input 1 :  
10  
Sample Output 1 :  
Positive  
Sample Input 2 :  
-10  
Sample Output 2 :  
Negative  

In [6]:
n = int(input())
if (n > 0):
    print("Positive")
elif (n == 0):
    print("Zero")
else:
    print("Negative")

Negative


### Question 2

#### Sum of n numbers

Given an integer n, find and print the sum of numbers from 1 to n.  
Note : Use while loop only.  

Input Format :  
Integer n  
Output Format :  
Sum  

Constraints :  
1 <= n <= 100

Sample Input :  
10  
Sample Output :  
55  

In [7]:
n  = int(input())
ctr = 1
sum = 0
while ctr <= n:
    sum = sum + ctr
    ctr = ctr + 1
print(sum)  

55


### Quesstion 3

#### Sum of Even Numbers

Given a number N, print sum of all even numbers from 1 to N.  

Input Format :  
Integer N  
Output Format :  
Required Sum   

Sample Input 1 :  
 6  
Sample Output 1 :  
12

In [1]:
N = int(input())
i = 1
sum = 0
while i <= N:
    if (i % 2 ==0):
        sum = sum + i
        i = i + 1
    else:
        i = i + 1
print(sum)  

12


### Question 4

#### Prime Number

Check to see weather a given number n is prime or not.

In [1]:
n = int(input())
d = 2
flag = False
while d < n:
    if(n%d==0):
        flag = True
    d = d + 1

if flag:
    print("Not prime")
else:
    print("prime")

Not prime


### Question 5

Check all prime numbers from 2 to n


In [3]:
n = int(input())
k = 2

while(k <= n):
    d = 2
    flag = False
    while (d < k):
        if(k % d == 0):
            flag = True
        d = d + 1
    if not flag:
        print(k)
    k = k + 1

2
3
5
7
11
13
17
19
23
29
31
37
41
43
47
53
59
61
67
71
73
79
83
89
97


### Question 6

#### Fahrenheit to Celsius

Given three values - Start Fahrenheit Value (S), End Fahrenheit value (E) and Step Size (W), you need to convert all Fahrenheit values from Start to End at the gap of W, into their corresponding Celsius values and print the table.  \

Input Format :  
3 integers - S, E and W respectively   

Output Format :  
Fahrenheit to Celsius conversion table. One line for every Fahrenheit and corresponding Celsius value. On Fahrenheit value and its corresponding Celsius value should be separate by tab ("\t")  

Constraints :  
0 <= S <= 80  
S <= E <=  900  
0 <= W <= 40   

Sample Input 1:  
0   
100   
20  

Sample Output 1:  
0   -17  
20  -6  
40  4  
60  15  
80  26  
100 37  

Sample Input 2:  
20  
119  
13  

Sample Output 2:  
20  -6  
33  0  
46  7  
59  15  
72  22  
85  29  
98  36  
111 43  

*Explanation For Input 2:*  
We need need to start calculating the Celsius values for each of the Fahrenheit Value which starts from 20. So starting from 20 which is the given Fahrenheit start value, we need to compute its corresponding Celsius value which computes to -6. We print this information as <Fahrenheit Value> a tab space"\t" <Celsius Value> on each line for each step of 13 we take to get the next value of Fahrenheit and extend this idea till we reach the end that is till 119 in this case. You may or may not exactly land on the end value depending on the steps you are taking.

In [3]:
S = int(input())
E = int(input())
W = int(input())

while(S <= E):
    cel = (S - 32) * 5/9
    print(S," ",int(cel))
    S = S + W

0   -17
20   -6
40   4
60   15
80   26
100   37


### Questoin 7

#### Calculator

Write a program that performs the tasks of a simple calculator. The program should first take an integer as input and then based on that integer perform the task as given below. 

1. If the input is 1, then 2 integers are taken from the user and their sum is printed.  
2. If the input is 2, then 2 integers are taken from the user and their difference(1st number - 2nd number) is printed.  
3. If the input is 3, then 2 integers are taken from the user and their product is printed.  
4. If the input is 4, then 2 integers are taken from the user and the quotient obtained (on dividing 1st number by 2nd number) is printed.  
5. If the input is 5, then 2 integers are taken from the user and their remainder(1st number mod 2nd number) is printed.  
6. If the input is 6, then the program exits.  
7. For any other input, then print "Invalid Operation".

Note: Each answer in next line.  

Input format:
Take integers as input, in accordance to the description of the question. 

Constraints:
Time Limit: 1 second

Output format:
The output lines must be as prescribed in the description of the question.

Sample Input:
3  
1  
2  
4  
4  
2  
1  
3  
2  
7  
6  

Sample Output:
2  
2  
5  

Invalid Operation
Explanation of the sample input
The first number given is 3, so that means two more numbers will be given and we'll have to multiply them and show the result. The two numbers are 1 and 2. Their product is 2, so 2 is displayed first in the output. Similarly, all the numbers are processed in groups of three. The first number tells the operation and the next two numbers tell on which numbers the operation is done. This applies to numbers from 1 to 5. If the input is 6 (like it is at the end), two more numbers will NOT be provided, you simply have to exit the program. Also, if the input is any number except 1 to 6 (like 7 which is at the second last), then you simply have to print "Invalid Operation"

In [25]:
while(1):
    a = int(input())
    
    if (a ==1):
        x = int(input())
        y = int(input())
        print(x+y)
    elif(a == 2):
        x = int(input())
        y = int(input())
        print(x-y)
    elif(a == 3):
        x = int(input())
        y = int(input())
        print(x*y)
    elif(a == 4):
        x = int(input())
        y = int(input())
        print(int(x/y))
    elif(a == 5):
        x = int(input())
        y = int(input())
        print(x%y)
    elif(a == 6):
        break
    else:
        print("Invalid Operation")

1


### Question 8

Reverse of a number  

Write a program to generate the reverse of a given number N. Print the corresponding reverse number.  
Note : If a number has trailing zeros, then its reverse will not include them. For e.g., reverse of 10400 will be 401 instead of 00401.  

Input format :  
Integer N

Output format :  
Corresponding reverse number

Constraints:  
0 <= N < 10^8  

Sample Input 1 :  
1234

Sample Output 1 :  
4321

Sample Input 2 :  
1980

Sample Output 2 :  
891

In [21]:
def reverse(n):
    rev = 0
    while(n > 0):
        reminder = n % 10
        rev = (rev * 10) + reminder
        n = n // 10
    return rev 


n=int(input())
result = reverse(n)
print(result)

891


### Question 9


#### Palindrome number

Write a program to determine if given number is palindrome or not. Print true if it is palindrome, false otherwise.  
Palindrome are the numbers for which reverse is exactly same as the original one. For eg. 121  

Sample Input 1 :  
121  
Sample Output 1 :  
true

Sample Input 2 :  
1032  
Sample Output 2 :  
false

In [26]:
def checkPalindrome(num):
    temp = num
    rev = 0
    while(num > 0):
        reminder = num % 10
        rev = (rev * 10) + reminder
        num = num // 10
        
    if (temp == rev):
        return True
    else:
        return False
     
     
num = int(input())
isPalindrome = checkPalindrome(num)
if(isPalindrome):
	print('true')
else:
	print('false')


false


### Question 10


#### Sum of even & odd

Write a program to input an integer N and print the sum of all its even digits and sum of all its odd digits separately.  
Digits mean numbers, not the places! That is, if the given integer is "13245", even digits are 2 & 4 and odd digits are 1, 3 & 5.  

Input format :
 Integer N

Output format :
Sum_of_Even_Digits Sum_of_Odd_Digits
(Print first even sum and then odd sum separated by space)

Constraints
0 <= N <= 10^8

Sample Input 1:  
1234  
Sample Output 1:  
6 4

Sample Input 2:  
552245  
Sample Output 2:  
8 15

Explanation for Input 2:  
For the given input, the even digits are 2, 2 and 4 and if we take the sum of these digits it will come out to be 8(2 + 2 + 4) and similarly, if we look at the odd digits, they are, 5, 5 and 5 which makes a sum of 15(5 + 5 + 5). Hence the answer would be, 8(evenSum) <single space> 15(oddSum)

In [16]:
N = int(input())
# To parse through the whole string.
#remember to keep initialization such as even and odd here outside the loop.
even = 0
odd = 0
while(N > 0):
    rem = N % 10
    N = N // 10
    if (rem % 2 == 0):
        even = even + rem
    else:
        odd = odd + rem    
print(even," ", odd)
    

8   15


### Question 11

#### Nth Fibonacci Number

Nth term of Fibonacci series F(n), where F(n) is a function, is calculated using the following formula -  
    F(n) = F(n-1) + F(n-2),   
    Where, F(1) = F(2) = 1  
Provided N you have to find out the Nth Fibonacci Number.  

Input Format :  
The first line of each test case contains a real number ‘N’.  
Output Format :  
For each test case, return its equivalent Fibonacci number.  
Constraints:  
1 <= N <= 10000       
Where ‘N’ represents the number for which we have to find its equivalent Fibonacci number.  

Time Limit: 1 second  
Sample Input 1:  
6  
Sample Output 1:  
8  

Explanation of Sample Input 1:  
Now the number is ‘6’ so we have to find the “6th” Fibonacci number  
So by using the property of the Fibonacci series i.e   

[ 1, 1, 2, 3, 5, 8]  
So the “6th” element is “8” hence we get the output.

In [2]:
N = int(input())
N1 = 0
N2 = 1
i = 1
if(N == 1):
    print(1)
else:
    while(i<N):
        Nth = N1 +N2
        N1 = N2
        N2 = Nth
        i += 1
    print(Nt) 
    

8


In [1]:
n = int(input())
a = 0
b = 1

c = -1
for i in range(n):
    c = a + b
    a = b
    b = c
print(a)

8


## Patterns 1

### Example

'****  
'****  
'****  
'****  

In [9]:
n = int(input())
i = 1
while i <= n:
    j = 1
    while j <= n:
        print("*",end="")
        j = j + 1
    print()
    i = i + 1


**
**


### Questoin 1

Code : Square Pattern

Print the following pattern for the given N number of rows.
Pattern for N = 4  
4444  
4444  
4444  
4444  

Input format :  
Integer N (Total no. of rows)  

Output format :
Pattern in N lines

Constraints  
0 <= N <= 50  

Sample Input 1:    
7  
Sample Output 1:  
7777777  
7777777  
7777777  
7777777  
7777777  
7777777  
7777777  

Sample Input 1:  
6

Sample Output 1:  
666666  
666666  
666666  
666666  
666666  
666666  

In [21]:
N = int(input())
i = 1
while i <= N:
    j = 1
    while j <= N:
        print(N,end = '')
        j = j+1
    print()
    i +=1

1111
2222
3333
4444


### Square patterns

In [22]:
N = int(input())
i = 1
while i<= N:
    j = 1
    while j <= N:
        print(i,end='')
        j += 1
    print()
    i += 1

1111
2222
3333
4444


In [23]:
N = int(input())
i = 1
while i<= N:
    j = 1
    while j <= N:
        print(j,end='')
        j += 1
    print()
    i += 1

1234
1234
1234
1234


In [24]:
N = int(input())
i = 1
while i<= N:
    j = 1
    while j <= N:
        print(N- j + 1,end='')
        j += 1
    print()
    i += 1

4321
4321
4321
4321


### Triangle Pattern

In [27]:
N = int(input())
i = 1
while i<= N:
    j = 1
    while j <= i:
        print(j,end='')
        j += 1
    print()
    i += 1

1
12
123
1234


In [29]:
n = int(input())
i = 1
while i <= n:
    j = 1
    p = i
    while j <= i:
        print(p, end="")
        j = j+1
        p = p+1
    print()
    i = i + 1  
        

1
23
345
4567


In [36]:
# In previos question we reset p now we have it globally
n = int(input())
i =1
p =1
while i<=n:
    j =1
    while j <=i:
        print(p,end ='')
        p = p +1
        j =j+1
    print()
    i = i + 1

1
23
456
78910


### Question 2

Code : Triangular Star Pattern  

Print the following pattern for the given N number of rows.  
Pattern for N = 4  
'*  
'**  
'***  
'****  

Note : There are no spaces between the stars (*).

Input format :
Integer N (Total no. of rows)

Output format :
Pattern in N lines

Constraints
0 <= N <= 50

Sample Input 1:  
5  
Sample Output 1:  
'*  
'**  
'***  
'****  
'*****  
Sample Input 2:  
6  
Sample Output 2:  
'*  
'**  
'***  
'****  
'*****  
'******

In [37]:
n = int(input())
i = 1
while i <= n:
    j = 1
    while j <= i:
        print("*", end='')
        j += 1
    print()
    i += 1

    

*
**
***
****


### Question 3

Code : Triangle Number Pattern

Print the following pattern for the given N number of rows.  
Pattern for N = 4  

1  
22  
333  
4444  

Input format :  
Integer N (Total no. of rows)

Output format : 
Pattern in N lines  

Constraints 
0 <= N <= 50

Sample Input 1: 
5   
Sample Output 1:  

1   
22  
333  
4444    
55555   

Sample Input 2: 
6   
Sample Output 2:  
  
1   
22  
333  
4444    
55555   
666666  

In [38]:
n = int(input())
i =1
while i <= n:
    j = 1
    while j <= i:
        print(i,end='')
        j += 1
    print()
    i +=1

1
22
333
4444


Code : Reverse Number Pattern

Print the following pattern for the given N number of rows. 
Pattern for N = 4   

1   
21  
321 
4321    

Input format :  
Integer N (Total no. of rows)

Output format :
Pattern in N lines  

Constraints  
0 <= N <= 50  

Sample Input 1:  
5  
Sample Output 1:  

1  
21  
321  
4321    
54321   
Sample Input 2:  
6   
Sample Output 2:    
1   
21  
321  
4321    
54321   
654321  

### Question 4

In [42]:
n = int(input())
i =1
while i <= n:
    j = 1
    p = i
    while j <= i:
        print(p,end='')
        j += 1
        p = p -1
    print()
    i +=1

1
21
321
4321
54321
654321


### Printing kth term

In [26]:
k = int (input())
# 'A' +k - 1
x = ord ('A')
assci_target = x +  k - 1
target_char = chr(assci_target)
print (target_char)

E


ABCD    
ABCD    
ABCD    
ABCD

In [15]:
n = int(input())
i = 1
while (i <= n):
    j = 1
    while( j <= n):
        charP = chr(ord('A') + j - 1) # to get the character.
        print(charP,end='')
        j += 1
    print()
    i += 1

ABCD
ABCD
ABCD
ABCD


ABCD    
BCDE    
CDEF    
DEFG

In [47]:
n = int(input())
i = 1
while (i <= n):
    j = 1
    start_char = chr(ord('A') + i - 1)
    while( j <= n):
        charP = chr(ord(start_char) + j - 1) # to get the character.
        print(charP,end='')
        j += 1
    print()
    i += 1

ABCD
BCDE
CDEF
DEFG


### Question 5

#### Character Pattern

Print the following pattern for the given N number of rows.  
Pattern for N = 4  

A   
BC  
CDE    
DEFG  

Input format :      
Integer N (Total no. of rows)   

Output format : 
Pattern in N lines

Constraints 
0 <= N <= 13

Sample Input 1: 
5   
Sample Output 1:   

A   
BC  
CDE  
DEFG    
EFGHI   

Sample Input 2:   
6   

Sample Output 2:    

A   
BC  
CDE  
DEFG    
EFGHI   
FGHIJK   

In [13]:
n = int(input())
i = 1
while i <= n :
    j = 1
    start_char = chr(ord('A') + i -1)
     
    while j <= i:
        charP = chr(ord(start_char) + j -1)
        print(charP, end = '')
        j += 1
    print()
    i += 1
    

A
BC
CDE
DEFG


In [44]:
# Same solution can also be done by no chr in earlier stages
n = int(input()) 
currRow = 1 
while currRow <= n:
    currCol = 1 
    ch = ord('A') + currRow - 1 
    while currCol <= currRow: 
        print(chr(ch + currCol - 1), end = "")
        currCol += 1 
    print() 
    currRow += 1

A
BC
CDE
DEFG


### Question 6

#### Pattern for N = 5  

E  
DE  
CDE  
BCDE  
ABCDE

In [36]:
n = int(input())
i = 1
while i <= n :
    j = 1
    start_char = chr(ord('A') + n -1)
     
    while j <= i:
        charP = chr(ord(start_char)-i +j)
        print(charP, end = '')
        j += 1
    print()
    i += 1

E
DE
CDE
BCDE
ABCDE


In [43]:
n = int(input())
i = 1 
while i <= n:
    j =1
    ch = ord('A') + n - i
    while j <=i:
        print(chr(ch+j-1), end ='')
        j +=1   
    print()
    i +=1

E
DE
CDE
BCDE
ABCDE


### Question 7

#### Number Pattern 1

Print the following pattern for the given N number of rows.  
Pattern for N = 4  
1  
11   
111   
1111 

In [48]:
n = int(input())
i = 1
while i<=n:
    j = 1
    while j<=i:
        print('1',end='')
        j +=1
    print()
    i +=1

1
11
111
1111


### Question 8

Number Pattern 2

Print the following pattern for the given N number of rows.  
Pattern for N = 4

1  
11  
121  
1221  

Input format :  
Integer N (Total no. of rows)

Output format : 
Pattern in N lines

Sample Input :  
5 

Sample Output :  
1  
11  
121  
1221  
12221  


In [11]:
n = int(input())
i = 1
print("1")
while (i <=n-1):
    j = 1
    print("1",end = '')
    while (j<= i-1):
        print("2",end = '')
        j +=1
    print("1", end = '')
    print()
    i +=1


1
11
121
1221
12221
122221


In [12]:
n = int(input())
i = 0
while (i <n):
    j = 0
    while (j <i+1):
        if(j ==0 or j ==i):
            print(1,end='')
        else:
            print(2,end ='')
        j += 1
    print()
    i+=1
    

1
11
121
1221


### Question 9


Number Pattern 3

Print the following pattern for the given N number of rows.  
Pattern for N = 4  
1  
11  
202  
3003    

Input format :  
Integer N (Total no. of rows)

Contraints:  
1 <= n <= 50

Output format :  
Pattern in N lines

Sample Input :  
5

Sample Output :  
1  
11  
202  
3003  
40004

In [2]:
n = int(input())
i = 1
print("1")
while (i <=n-1):
    j = 1
    print(i,end = '')
    while (j<= i-1):
        print("0",end = '')
        j +=1
    print(i, end = '')
    print()
    i +=1

1
11
202
3003
40004


In [5]:
n = int(input())
print('1')
i = 1
while (i <n):
    j = 0
    while (j <i+1):
        if(j ==0 or j ==i):
            print(i,end='')
        else:
            print(0,end ='')
        j += 1
    print()
    i+=1

1
11
202
3003
40004


### Question 10

Number Pattern

Print the following pattern for the given N number of rows.  
Pattern for N = 4  

1234  
123  
12  
1  

Input format :  
Integer N (Total no. of rows)

Output format :  
Pattern in N lines

Sample Input :  
5

Sample Output :

12345  
1234  
123  
12  
1  

In [50]:
n = int(input())
i = 1
while i<=n:
    j = 1
    while j<=(n-i+1):
        print(j,end='') 
        j +=1
    print()
    i +=1

1234
123
12
1


### Questoin 11

Alpha Pattern

Print the following pattern for the given N number of rows.  
Pattern for N = 3  

 A  
 BB  
 CCC

Input format :  
Integer N (Total no. of rows)

Output format :  
Pattern in N lines

Constraints  
0 <= N <= 26

Sample Input 1:  
7

Sample Output 1:  

A  
BB  
CCC  
DDDD  
EEEEE  
FFFFFF  
GGGGGGG  

Sample Input 2:  
6

Sample Output 2:  

A  
BB  
CCC  
DDDD  
EEEEE  
FFFFFF  

In [9]:
n = int(input())
i = 1
while (i <=n):
    start_char = chr(ord('A') + i -1)
    j =1
    while (j <=i):
        print(start_char,end='')
        j += 1
    print()
    i += 1

A
BB
CCC
DDDD


## Patterns 2